### PR TITLE
adding unit test for image.load_extended()

### PR DESCRIPTION
**Pull request by kristofszentes**
I added a unit test for image.load_extended(), and some small images (smaller than 32x32 pixels) in the example/data folder to be able to test all the following formats: BMP, PNG, JPG, GIF (non-animated), TGA (uncompressed), TIF, XPM, PPM, PGM, as it was asked in [this previous attempt](https://github.com/pygame/pygame/pull/2014) of a such unit test.
[Here](https://github.com/pygame/pygame/issues/1778) is the issue related to this unit test.

### CODE REVIEW

File: test/image_test.py [hunk 1]
```diff
@@ -806,13 +806,55 @@ def test_load_basic(self):
...
 
         # __doc__ (as of 2008-08-02) for pygame.image.load_extended:
 
         # pygame module for image transfer
 
-        self.fail()
+        #We test loading the following file types: bmp, png, jpg, gif (non-animated), tga (uncompressed), tif, xpm, ppm, pgm
+        #All the loaded images are smaller than 32 x 32 pixels
+
+        #test loading a bmp
```
Comment (illume) [inline test/image_test.py:818]: We only recently added support for using subTest inside the pygame test suite, but they're excellent! With them you can make the messages better if something fails. Also it allows the further parts to be tested if something at the top of the test fails (for example if the png loading fails all the subTests after that will run). Have a look at the docs for subTest: https://docs.python.org/3/library/unittest.html#subtests

Something similar to this... (untested):
```python
with self.subTest("test loading a bmp", filename=filename, expected_color=(...)):
    surf = pygame.image.load_extended(example_path(filename))
    self.assertEqual(surf.get_at((0,0)), expected_color)
```

It's also possible to loop over a data structure of [[filename, expected_color], ...]

**Review by illume — APPROVED** [review]
👍 thank you :) It's great to have tests for all those file formats now.

We could merge this in if you're happy with it. I left a note about subTest, which you may want to look into. Another minor thing is that the spacing in the colors could be improved to be pep8 friendly.

**Comment by kristofszentes** [comment]
Thank you for your advice! 
I changed spacing for colors, added the use of subTest and a for loop going through all the tests.

**Comment by illume** [comment]
Brilliant. Thanks again.

**Pull request merged**